java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -551,4 +556,1697 @@ public void runPiranhaWithFewProperties() throws IOException {
       assert_().fail("Incorrect parameters passed to the checker");
     }
   }
+
+  /*
+   * Test for the case when xpFlagName is ""
+   * Methods with no arguments are simplified, provided argumentIndex is not specified
+   * (assuming that returnType and receiverType - if specified - are a match)
+   * Uses "properties_test_noFlag.json" as the config file.
+   * Note that "Piranha:FlagName" is not considered when argumentIndex is not specified.
```

Comment:
Out of curiosity: will this work only for versions of the method that take zero args?

My worry here is that a misconfiguration, where `Piranha:FlagName` is empty, and the `argumentIndex` is missing, it will go and delete half the code. We probably don't want that behavior. @murali ??

Reply:
I think we'd want any code that would delete `experimentation.isToggleEnabled(SOME_FLAG) { ... }` for every possible value of `SOME_FLAG` at once to error out by default. If someone really needs to do this, we can add a `Piranha:AllowMassRemoval=true` argument or something. It just feels like the kind of thing that might scare new users from trying piranha if they don't configure it right. 

Unless I am mistaken and this case will only delete treatment methods with zero args.

Reply:
My understanding is that this will only delete methods listed in the properties.json without checking for flagname in argument. At worst, this will delete all calls to an API if the argumentIndex is not specified. Your point on getting the configuration right is valid too. 

Let us leave this functionality as it is in the current version of the PR and if it becomes a source of confusion, let us revisit this.

Reply:
I feel that worst case scenario is really really bad. Consider the effects of trying out Piranha in Uber's monorepos under those conditions, it would locally rewrite maybe hundreds of thousands of lines of code... and scare many people from ever trying it again.

I am fine with that check being a follow-up task, but I'd want to have it in place before we cut any release with this (it's already tricky enough that we are switching the config file format)

Reply:
Also for a follow up task (I can do it): Print a message whenever the config points to a file ending in `.properties` explaining the format change and linking the docs in case someone is just upgrading Piranha.

Reply:
How about this: 
a) By default, argumentIndex is a required parameter in json. 
b) Using a configuration option (as you point above), this can be made optional. 

(a) will be good for most users and in cases, where a user wants to delete all calls to the API, (b) can be used.

Reply:
Yup. That's exactly what I'd want. And we can add (a) to this PR and make (b) a follow up, to make this easier to land. But I'd rather have that small guardrail there than the extra functionality of deleting methods without arguments 😄

Reply:
^ @pranavsb Everything else is a nit and could be done later, but I'd like this one change before landing:

> a) By default, argumentIndex is a required parameter in json.

Reply:
>Also for a follow up task (I can do it): Print a message whenever the config points to a file ending in .properties explaining the format change and linking the docs in case someone is just upgrading Piranha.

Raised #49 for this. Clearly, I have issues :P